src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -0,0 +1,116 @@
+import org.mindrot.jbcrypt.BCrypt;
+import java.util.UUID;
+
+
+/** Servlet class responsible for the Admin page. */
+public class AdminServlet extends HttpServlet {
+
+  private UserStore userStore;
+  private ConversationStore conversationStore;
+  private MessageStore messageStore;
+  private List<String> adminList = new ArrayList<>();
+  /**
+   * Set up state for handling login-related requests. This method is only called when running in a
+   * server, not when running in a test.
+   */
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(userStore.getInstance());
+    setConversationStore(conversationStore.getInstance());
+    setMessageStore(messageStore.getInstance());
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+  /**
+   * Sets the ConversationStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for
+   * use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+      String username = (String)request.getSession().getAttribute("user");
+      if (username == null) {
+        // user is not logged in, don't let them access the admin page
+        response.sendRedirect("/");
+        return;
+      }
+      User user = userStore.getUser(username);
+
+      if (user == null) {
+        // user was not found, don't let them access the admin page
+        response.sendRedirect("/");
+        return;
+      }
+      adminList.add("admin");
+      if (user.getIsAdmin() == true) {
+        if (!adminList.contains(user.getName())){
+          adminList.add(user.getName());
+        }
+      }
+
+      request.setAttribute("adminList", adminList);
+      if (adminList.contains(user.getName())) {
+      /* an attempt to grab information from the stores to display on the page
+      if the user is admin*/
+          List<User> userList = UserStore.getInstance().getAllUsers();
+          int numOfUsers = userList.size();
+          int numOfConvos = conversationStore.getInstance().getAllConversations().size();
+          int numOfMessages = messageStore.getInstance().getAllMessages().size();
+          String newestUser = userList.get(userList.size()-1).getName();
+          String mostActiveUser = "";
```

Comment:
When the app is launched, `mostActiveUser` will be `""`. If that's the case, maybe we can display a message encouraging users to be active.

Reply:
I can do that